### PR TITLE
Adding rendering of fermata elements

### PR DESCRIPTION
**Pull request by rettinghaus**
It works! Looks fine so far. But like with `<pedal>` it doesn't work with `@startid`.

### CODE REVIEW

**Comment by lpugin** [comment]
Great! I'll look at it. The `@startid` issue should be fixed with the latest push in develop

File: src/view_floating.cpp [hunk 1]
```diff
@@ -58,9 +59,9 @@ void View::DrawControlElement(DeviceContext *dc, ControlElement *element, Measur
     assert(measure);
     assert(element);
 
-    // For dir, dynam, and harm, we do not consider the @tstamp2 for rendering
+    // For dir, dynam, fermata, and harm, we do not consider the @tstamp2 for rendering
     if (element->HasInterface(INTERFACE_TIME_SPANNING) && (element->Is() != DIR) && (element->Is() != DYNAM)
-        && (element->Is() != HARM)) {
+        && (element->Is() != FERMATA) && (element->Is() != HARM)) {
```
Comment (lpugin) [inline src/view_floating.cpp]: I am not sure this is necessary because a fermata has no TIME_SPANNING interface anyway

File: src/system.cpp [hunk 2]
```diff
@@ -325,6 +325,8 @@ int System::AdjustFloatingPostioners(FunctorParams *functorParams)
     m_systemAligner.Process(params->m_functor, params);
     params->m_classId = PEDAL;
     m_systemAligner.Process(params->m_functor, params);
+    params->m_classId = FERMATA;
```
Comment (lpugin) [inline src/system.cpp]: I would move this before OCTAVE (lines 316-7)

File: src/view_floating.cpp [hunk 3]
```diff
@@ -1533,14 +1539,58 @@ void View::DrawDynam(DeviceContext *dc, Dynam *dynam, Measure *measure, System *
...
+    str.push_back(code);
+    
+    std::vector<Staff *>::iterator staffIter;
+    std::vector<Staff *> staffList = fermata->GetTstampStaves(measure);
+    for (staffIter = staffList.begin(); staffIter != staffList.end(); staffIter++) {
+        system->SetCurrentFloatingPositioner((*staffIter)->GetN(), fermata, x, (*staffIter)->GetDrawingY());
+        // Basic method that use bounding box
+        int y = fermata->GetDrawingY();
+        
+        // Adjust the x position differently for up and down
```
Comment (lpugin) [inline src/view_floating.cpp]: This comment does not make sense

File: src/view_floating.cpp [hunk 4]
```diff
@@ -1533,14 +1539,58 @@ void View::DrawDynam(DeviceContext *dc, Dynam *dynam, Measure *measure, System *
...
+
 void View::DrawHarm(DeviceContext *dc, Harm *harm, Measure *measure, System *system)
 {
     assert(dc);
     assert(system);
     assert(measure);
     assert(harm);
 
-    // We cannot draw a dir that has no start position
+    // We cannot draw a harmonic that has no start position
```
Comment (lpugin) [inline src/view_floating.cpp]: This was wrong indeed. I'd say "harmony indication", though.

File: src/view_floating.cpp (hunk 3 above)
Comment (rettinghaus) [inline src/view_floating.cpp]: copy & paste error

**Comment by rettinghaus** [comment]
fermatas

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title />
            </titleStmt>
            <pubStmt>
                <date>2016-11-21 14:23:25</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Encoded with Verovio version 0.9.13-dev-9aae34a-dirty</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000194503874">
                        <staffGrp xml:id="m-000000050309315">
                            <staffDef xml:id="staffdef-000000001593092" clef.shape="G" clef.line="2" key.sig="1s" meter.count="4" meter.unit="4" n="1" label="Piano" label.abbr="Pno." lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section xml:id="section-000000196269157">
                        <measure xml:id="measure-000000165135762" n="1">
                            <staff xml:id="staff-000000103277373" n="1">
                                <layer xml:id="layer-000000039278325" n="1">
                                    <note xml:id="note-000000121464704" dur="2" oct="4" pname="g" stem.dir="up" />
                                    <note xml:id="note-000000209770470" dur="2" oct="4" pname="g" stem.dir="up" />
                                </layer>
                            </staff>
                            <fermata tstamp="1" staff="1" place="above"/>
                            <fermata tstamp="3" staff="1" place="below"/>
                        </measure>
                        <measure xml:id="measure-000000187211871" n="2">
                            <staff xml:id="staff-000000074744283" n="1">
                                <layer xml:id="layer-000000084122809" n="1">
                                    <note xml:id="note-000000102928941" dur="2" oct="4" pname="g" stem.dir="up" />
                                    <note xml:id="note-000000196906090" dur="2" oct="4" pname="g" stem.dir="up" />
                                </layer>
                            </staff>
                            <fermata tstamp="1" staff="1" place="above" form="inv"/>
                            <fermata tstamp="3" staff="1" place="below" form="norm"/>
                        </measure>
                        <measure xml:id="measure-000000182813266" n="3">
                            <staff xml:id="staff-000000158182391" n="1">
                                <layer xml:id="layer-000000054425405" n="1">
                                    <note xml:id="note-000000056394863" dur="2" oct="4" pname="g" stem.dir="up" />
                                    <note xml:id="note-000000159775484" dur="2" oct="4" pname="g" stem.dir="up" />
                                </layer>
                            </staff>
                            <fermata tstamp="1" staff="1"/>
                            <fermata tstamp="1" staff="1" place="below"/>
                        </measure>
                        <measure xml:id="measure-000000015635268" n="4" right="end">
                            <staff xml:id="staff-000000168915082" n="1">
                                <layer xml:id="layer-000000124815953" n="1">
                                    <note xml:id="note-000000046622582" dur="2" oct="4" pname="g" stem.dir="up" />
                                    <note xml:id="note-000000209755930" dur="2" oct="4" pname="g" stem.dir="up" />
                                </layer>
                            </staff>
                            <fermata startid="#note-000000046622582"/>
                            <fermata startid="#note-000000209755930"/>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
Does `@startid` works if `@staff` is provided?

**Comment by rettinghaus** [comment]
No, it doesn't. Neither does `<pedal>`.

**Comment by lpugin** [comment]
Even after merging what I pushed yesterday to develop?

**Comment by rettinghaus** [comment]
Yeah! That did the trick! I merged it and now everything works just fine.

**Comment by rettinghaus** [comment]
If you'll add glyphs for squared and angular fermatas, I'll add them, too.

**Comment by lpugin** [comment]
Will do later this week

**Comment by lpugin** [comment]
I added more glyphs in the last commit in develop

**Pull request merged**